orchagent/crmorch.cpp

```diff
@@ -429,11 +450,17 @@ void CrmOrch::getResAvailableCounters()
             case SAI_SWITCH_ATTR_AVAILABLE_NEXT_HOP_GROUP_MEMBER_ENTRY:
             case SAI_SWITCH_ATTR_AVAILABLE_NEXT_HOP_GROUP_ENTRY:
             case SAI_SWITCH_ATTR_AVAILABLE_FDB_ENTRY:
+            case SAI_SWITCH_ATTR_AVAILABLE_IPMC_ENTRY:
+            case SAI_SWITCH_ATTR_AVAILABLE_SNAT_ENTRY:
+            case SAI_SWITCH_ATTR_AVAILABLE_DNAT_ENTRY:
             {
                 sai_status_t status = sai_switch_api->get_switch_attribute(gSwitchId, 1, &attr);
                 if (status != SAI_STATUS_SUCCESS)
                 {
-                    SWSS_LOG_ERROR("Failed to get switch attribute %u , rv:%d", attr.id, status);
+                    if(status != SAI_STATUS_NOT_SUPPORTED)
+                    {
+                        SWSS_LOG_ERROR("Failed to get switch attribute %u , rv:%d", attr.id, status);
```

Comment:
if it is not supported, i think it is better to remove query. otherwise, in the syslog, we are going to continously having this error log for this unsupported attribute. it will cause confusion.

Reply:
you mean to remove it from m_resourcesMap on the first failure with SAI_STATUS_NOT_SUPPORTED?

Reply:
yes

Reply:
please note that the default poll interval is 5 minutes and the logs will be observed once every 5 minutes (poll interval)

Reply:
@PrabhuSreenivasan , we change the polling interval to 1 sec while running tests and this can come pretty often in that case causing test failures.

Reply:
@lguohan, @prsunny, I have updated the code to remove an unavailable resource from the map.